tests/activity_api_pause_test.go

```diff
@@ -0,0 +1,464 @@
+
+	s.initialRetryInterval = 1 * time.Second
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+	}
+
+	var activityCompleted atomic.Int32
+
+	activityFunction := func() (string, error) {
+		if activityCompleted.Load() == 0 {
+			activityErr := errors.New("bad-luck-please-retry")
+			activityCompleted.Add(1)
+			return "", activityErr
+		}
+		activityCompleted.Add(1)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        testcore.RandomizeStr("wf_id-" + s.T().Name()),
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+
+	// wait for activity to start
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, int32(1), activityCompleted.Load())
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// pause activity
+	pauseRequest := &workflowservicepb.PauseActivityByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+	}
+	resp, err := s.FrontendClient().PauseActivityById(ctx, pauseRequest)
+	s.NoError(err)
+	s.NotNil(resp)
+
+	// wait long enough for activity to retry if pause is not working
+	util.InterruptibleSleep(ctx, 2*time.Second)
+
+	// make sure activity is not completed, and was not retried
+	description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+	s.NoError(err)
+	s.Equal(1, len(description.PendingActivities))
+	s.True(description.PendingActivities[0].Paused)
+	s.Equal(int32(2), description.PendingActivities[0].Attempt)
+
+	// unpause the activity
+	unpauseRequest := &workflowservicepb.UnpauseActivityByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+		Operation: &workflowservicepb.UnpauseActivityByIdRequest_Resume{
+			Resume: &workflowservicepb.UnpauseActivityByIdRequest_ResumeOperation{
+				NoWait: false,
+			},
+		},
+	}
+	unpauseResp, err := s.FrontendClient().UnpauseActivityById(ctx, unpauseRequest)
+	s.NoError(err)
+	s.NotNil(unpauseResp)
+
+	// wait for activity to complete
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		assert.Equal(t, int32(2), activityCompleted.Load())
+	}, 5*time.Second, 100*time.Millisecond)
+
+	var out string
+	err = workflowRun.Get(ctx, &out)
+
+	s.NoError(err)
+
+}
+
+func (s *ActivityApiPauseClientTestSuite) TestActivityPauseApi_WhileRetryNoWait() {
+	// in this case pause happen when activity for in retry state
+	// make sure that activity is paused and then unpaused
+	// also test noWait flag
```

Comment:
```suggestion
	// In this case, pause can happen when activity is in retry state.
	// Make sure that activity is paused and then unpaused.
	// Also tests noWait flag.
```

(easier to read with punctuation)